src/gn.cu

```diff
@@ -286,17 +285,21 @@ static double cuda_score_host(network_t gpu_n, const experiment_set_t gpu_eset,
   // cudaMemcpy(gpu_s_kernels, s_kernels, N*sizeof(double), cudaMemcpyHostToDevice);
 
   // make grid and block sizes according to input
-  int TILE = 8;
-  dim3 dimGrid((N - 1)/TILE + 1, (N - 1)/TILE + 1);
-  dim3 dimBlock(TILE, 1, 1);
-  // launch kernel
-  cuda_score_device<<<dimGrid, dimBlock>>>(N, gpu_n, gpu_eset, gpu_trajectories, limit, max_states, gpu_s_kernels);
+  // int TILE = 8;
+  // dim3 dimGrid((N - 1)/TILE + 1, (N - 1)/TILE + 1);
+  // dim3 dimBlock(TILE, 1, 1);
+  // cuda_score_device<<<dimGrid, dimBlock>>>(N, gpu_n, gpu_eset, gpu_trajectories, limit, max_states, gpu_s_kernels);
+
+  // Using 1D blocks
+  int numThreads = 32;
+  int numBlocks = (N + numThreads - 1 )/ numThreads;
+  cuda_score_device<<<numBlocks, numThreads>>>(N, gpu_n, gpu_eset, gpu_trajectories, limit, max_states, gpu_s_kernels);
```

Comment:
The dimGrid approach wasn't correct as we are using an one dimensional block grid and not 2D which caused multiple GPU threads to pick up the same experiment